pennylane/devices/default_mixed.py

```diff
@@ -494,6 +508,22 @@ def _apply_operation(self, operation):
             self._apply_channel(matrices, wires)
 
     # pylint: disable=arguments-differ
+
+    def statistics(self, observables, shot_range=None, bin_size=None):
+        """Applies a readout error to the measurement outcomes of any observable if
+        readout_prob is non-zero. It applies to the following properties being
+        measured: Expectation, Variance, Sample, and Probability. This is done
```

Comment:
Note: we also have a Counts measurement process now (available via `qml.sample(..., counts=True)`. Would the readout error be working there too?

Reply:
Hi @antalszava, 
I realized that this way of defining readout error using `statistics` method rather than the `apply` method seem to work only for exact calculations when `shots` is not given by the user.  When `shots` is explicitly specified, then the results of measurements of observables don't seem to incorporate the readout error. I am trying to see why this happens. After than, I can check if the readout error is working with the Counts measurement process or not.

Reply:
Hi @antalszava, thanks a lot for the detailed explanation :) That's very helpful! Here is a working version of the code:
```python
def execute(self, circuit, **kwargs):
        if self.readout_err:
            for obs in circuit.observables:
                if obs.return_type in (State,VnEntropy,MutualInfo):
                    self.measured_wires = []
                    return super().execute(circuit, **kwargs)
                elif obs.return_type in (Sample,Counts):
                    if obs.name == "Identity":
                        self.measured_wires = self.wires
                        return super().execute(circuit, **kwargs)
            self.measured_wires = qml.wires.Wires.all_wires([obs.wires for obs in circuit.observables])
        return super().execute(circuit, **kwargs)
```
This version of code should also resolve https://github.com/PennyLaneAI/pennylane/pull/2786#discussion_r921212266. Please let me know your thoughts.